### PR TITLE
[Repository Signing] Leave nuget.org's repository signature

### PR DESCRIPTION
**Pull request by loic-sharma**
See https://github.com/NuGet/Engineering/issues/1327

**NOTE**: This change will report validation issues to the customer if nuget.org's repository signature fails trust or integrity verification. This is not desirable. Instead, nuget.org's invalid repository signature should be stripped so that another one can be applied. This will be fixed by https://github.com/NuGet/Engineering/issues/1401

### CODE REVIEW

File: src/Validation.PackageSigning.ProcessSignature/ISignatureFormatValidator.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Packaging.Signing;
+
+namespace NuGet.Jobs.Validation.PackageSigning.ProcessSignature
+{
+    public interface ISignatureFormatValidator
+    {
+        Task<VerifySignaturesResult> ValidateMinimalAsync(
```
Comment (skofman1) [inline src/Validation.PackageSigning.ProcessSignature/ISignatureFormatValidator.cs]: add comments

File: src/Validation.PackageSigning.ProcessSignature/Settings/dev.json [hunk 2]
```diff
@@ -17,6 +17,12 @@
   "ValidationStorage": {
     "ConnectionString": "DefaultEndpointsProtocol=https;AccountName=nugetdevlegacy;AccountKey=$$Dev-NuGetDevLegacyStorage-Key$$"
   },
+  "ProcessSignature": {
+    "AllowedRepositorySigningCertificates": [
+      "3f9001ea83c560d712c24cf213c3d312cb3bff51ee89435d3430bd06b5d0eece"
```
Comment (dtivel) [inline src/Validation.PackageSigning.ProcessSignature/Settings/dev.json]: This is the thumbprint for the Microsoft **author** signing certificate.  The thumbprint for the NuGet.org repository signing certificate is `0E5F38F57DC1BCC806D8494F4F90FBCEDD988B46760709CBEEC6F4219AA6157D`.

File: src/Validation.PackageSigning.ProcessSignature/SignatureValidator.cs [hunk 3]
```diff
@@ -315,77 +337,181 @@ public class SignatureValidator : ISignatureValidator
...
+                    context.Message.PackageVersion,
+                    context.Message.ValidationId);
+
+                return false;
+            }
+
+            return true;
+        }
+
+        private bool IsValidRepositorySignature<T>(Context context, T signature)
```
Comment (dtivel) [inline src/Validation.PackageSigning.ProcessSignature/SignatureValidator.cs]: Nice!

File: src/Validation.PackageSigning.ProcessSignature/Settings/dev.json (hunk 2 above)
Comment (loic-sharma) [reply src/Validation.PackageSigning.ProcessSignature/Settings/dev.json]: Oops. I fixed the job, I'll also need to fix the v3 resource and our monitoring. Tracked by: https://github.com/NuGet/Engineering/issues/1402

File: src/Validation.PackageSigning.ProcessSignature/SignatureValidator.cs (hunk 3 above)
Comment (loic-sharma) [reply src/Validation.PackageSigning.ProcessSignature/SignatureValidator.cs]: Kudos to @joelverhagen for that one :)

**Comment by loic-sharma** [comment]
Tool to generate a repository signed package: https://gist.github.com/loic-sharma/e5c6af48e62607a30bd8dfdef4b02bc4
Tool to generate code signing certificate: https://gist.github.com/joelverhagen/0acace29e8341a4a954732b164cd4193

**Pull request merged**